eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -21,3 +21,178 @@ function Get-NamepspacesFromDll($dllPath) {
     }
     return $namespaces | Sort-Object -Unique
 }
+
+function DownloadNugetPackage($package, $version, $destination) {
+    # $PackageSourceOverride is a global variable provided in
+    # Update-DocsMsToc.ps1. Its value can set a "customSource" property.
+    # If it is empty then the property is not overridden.
+    $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+    # Download package from nuget, if no package found in nuget, then search devops public feeds if set.
+    Write-Host "Downloading $package with version $version..."
+    if ($customPackageSource) {
+        nuget install -FallbackSource $customPackageSource $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+    }
+    else {
+        nuget install $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+    }
+}
+
+function Fetch-NamespacesFromNupkg ($package, $version) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "extractNupkg")
+    if (!(Test-Path $tempLocation)) {
+        New-Item -ItemType Directory -Path $tempLocation -Force | Out-Null
+    }
+    Write-Host "Downloading nupkg package $package with version $version to $tempLocation ...."
+    DownloadNugetPackage -package $package -version $version -destination $tempLocation | Out-Null
+    $packageFolder = "$package.$version"
+    if ($LASTEXITCODE -ne 0 -or !$(Test-Path $tempLocation/$packageFolder)) {
+        Write-Host "Did not download the package correctly. Skipping..."
+        return @()
+    }
+    Write-Host "Searching for dll file..."
+    $dllFile = @()
+    if (Test-Path "$tempLocation/$packageFolder/lib/netstandard2.0/"){
+        $dllFile = @(Get-ChildItem "$tempLocation/$packageFolder/lib/netstandard2.0/*" -Filter '*.dll' -Recurse)
+    }
+    if (!$dllFile -or ($dllFile.Count -gt 1)) {
+        Write-Warning "Can't find any dll file from $tempLocation/$packageFolder with target netstandard2.0."
```

Comment:
Is this supposed to be a `-lt` check? If `-gt` the error message could be a bit confusing.

Reply:
Actually, the dllFile should only has one in each target framework (e.g. netstandard2.0). The ($dllFile.Count -gt 1) is redundant for normal world.

Reply:
If there are > 1 dll files, and the log warning says "can't find any dll file" I think that could be misleading.

Reply:
After the test, I found we do have package which includes multiple dlls in one target. Added elseif the the case.
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1971666&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=98c76ae6-047b-5869-f98c-bcf6099a854a&l=623 
<img width="1359" alt="image" src="https://user-images.githubusercontent.com/48036328/200417807-f87f167f-2c5f-4ca2-8863-6ac68c447af4.png">

```
/tmp/extractNupkg/Azure.Cosmos.4.0.0-preview3/lib/netstandard2.0/Azure.Cosmos.dll
/tmp/extractNupkg/Azure.Cosmos.4.0.0-preview3/lib/netstandard2.0/Microsoft.Azure.Cosmos.Core.dll 
/tmp/extractNupkg/Azure.Cosmos.4.0.0-preview3/lib/netstandard2.0/Microsoft.Azure.Cosmos.Direct.dll 
/tmp/extractNupkg/Azure.Cosmos.4.0.0-preview3/lib/netstandard2.0/Microsoft.Azure.Cosmos.Serialization.HybridRow.dll

```